syntax = "proto3";

package sigmiotdata;

message LogDataResponse {
  string log_message = 1;
  uint64 log_timestamp = 2;
  string log_level = 3;
}

message SensorValue {
  string value_name = 1;
  float value_data = 2;
  string value_unit = 3;
}

message SensorDataResponse {
  string sensor_name = 1;
  string sensor_type = 2;
  string sensor_location = 3;
  repeated SensorValue sensor_values = 4;
}

message MessageResponse {
  enum Status {
    OK = 0;
    ERR = 1;
    NOT_FOUND = 2;
  }
  Status status = 1;
  repeated SensorDataResponse sensors_data_response = 2;
  repeated LogDataResponse log_data_response = 3;
}
